### PR TITLE
Add domain mapping deployment in Serving e2e test

### PR DESCRIPTION
**Pull request by skonto**
See https://github.com/openshift/knative-serving/pull/1040#issuecomment-1023016720. Let's see if that fails in any case.
Related to #1385

### CODE REVIEW

File: test/e2e/knative_serving_test.go [hunk 1]
```diff
@@ -26,13 +26,13 @@ func TestKnativeServing(t *testing.T) {
 
 	t.Run("verify correct deployment shape", func(t *testing.T) {
 		// Check the desired scale of deployments in the knative serving namespace
-		for _, deployment := range []string{"activator", "controller", "autoscaler-hpa"} {
+		for _, deployment := range []string{"activator", "controller", "autoscaler-hpa", "domain-mapping"} {
```
Comment (nak3) [inline test/e2e/knative_serving_test.go:29]: We can add `domainmapping-webhook` and `webhook` here.

File: test/e2e/knative_serving_test.go [hunk 2]
```diff
@@ -26,13 +26,13 @@ func TestKnativeServing(t *testing.T) {
...
 			if err := test.CheckDeploymentScale(caCtx, servingNamespace, deployment, haReplicas); err != nil {
 				t.Fatalf("Failed to verify default HA settings for %q: %v", deployment, err)
 			}
 		}
 		// Check the status of deployments in the knative serving namespace
-		for _, deployment := range []string{"activator", "autoscaler", "autoscaler-hpa", "controller", "webhook"} {
+		for _, deployment := range []string{"activator", "autoscaler", "autoscaler-hpa", "controller", "domain-mapping", "webhook"} {
```
Comment (nak3) [inline test/e2e/knative_serving_test.go:35]: We can add `domainmapping-webhook`.

**Review by nak3 — COMMENTED** [review]
I think we can fix https://github.com/openshift-knative/serverless-operator/issues/1377 here but do you have a reason why we want to add `domain-mapping` only?

File: test/e2e/knative_serving_test.go (hunk 1 above)
Comment (skonto) [reply test/e2e/knative_serving_test.go]: Autoscaler has no ha settings?

Comment (nak3) [reply test/e2e/knative_serving_test.go]: Yeah, autoscaler has the HA setting. :sweat_smile: Let's add it.

**Comment by nak3** [comment]
By the way, as discussed https://github.com/openshift-knative/serverless-operator/pull/1376#discussion_r787495867 we can refactor these loops to one loop. Would you like to change it? Or another PR?

**Comment by skonto** [comment]
Yeah I was thinking about the loops, wanted to make sure first about the autoscaler's ha settings.

**Comment by skonto** [comment]
@nak3 ready.

File: test/e2e/knative_serving_test.go [hunk 3]
```diff
@@ -25,14 +25,16 @@ func TestKnativeServing(t *testing.T) {
...
 			if err := test.CheckDeploymentScale(caCtx, servingNamespace, deployment, haReplicas); err != nil {
 				t.Fatalf("Failed to verify default HA settings for %q: %v", deployment, err)
 			}
 		}
 		// Check the status of deployments in the knative serving namespace
-		for _, deployment := range []string{"activator", "autoscaler", "autoscaler-hpa", "controller", "webhook"} {
+		for _, deployment := range servingDeployments {
 			if _, err := test.WithDeploymentReady(caCtx, deployment, servingNamespace); err != nil {
 				t.Fatalf("Deployment %s is not ready: %v", deployment, err)
 			}
```
Comment (nak3) [inline test/e2e/knative_serving_test.go:40]: Um.. I assumed that your comment in https://github.com/openshift-knative/serverless-operator/pull/1376#discussion_r787495867 suggested the one "loop" as:

```go
	for _, deployment := range servingDeployments {
		// Check the desired scale of deployments in the knative serving namespace
		if err := test.CheckDeploymentScale(caCtx, servingNamespace, deployment, haReplicas); err != nil {
			t.Fatalf("Failed to verify default HA settings for %q: %v", deployment, err)
		}

		// Check the status of deployments in the knative serving namespace
		if _, err := test.WithDeploymentReady(caCtx, deployment, servingNamespace); err != nil {
			t.Fatalf("Deployment %s is not ready: %v", deployment, err)
		}
	}
```

But we don't need to change it? :sweat_smile: If we will make it one loop, we also need to change L#51 below, though.

Comment (skonto) [reply test/e2e/knative_serving_test.go:40]: Yes forgot to update it. So tests pass let me do that minor thingy.

File: test/e2e/knative_serving_test.go [hunk 4]
```diff
@@ -25,14 +25,15 @@ func TestKnativeServing(t *testing.T) {
 	})
 
 	t.Run("verify correct deployment shape", func(t *testing.T) {
-		// Check the desired scale of deployments in the knative serving namespace
-		for _, deployment := range []string{"activator", "controller", "autoscaler-hpa"} {
+		servingDeployments := []string{"activator", "autoscaler", "autoscaler-hpa", "controller", "domain-mapping", "domainmapping-webhook", "webhook"}
+
+		for _, deployment := range servingDeployments {
```
Comment (nak3) [inline test/e2e/knative_serving_test.go:30]: When we refactor this loop, it is better to refactor the same codes in other places :sweat_drops: 
For example:
https://github.com/openshift-knative/serverless-operator/blob/ed2b76b759db261925e0fc6bd0d1788dc18b69e6/test/e2e/knative_serving_test.go#L51-L61
https://github.com/openshift-knative/serverless-operator/blob/ed2b76b759db261925e0fc6bd0d1788dc18b69e6/test/e2e/knative_eventing_test.go#L39-L49

Comment (skonto) [reply test/e2e/knative_serving_test.go:30]: Not sure I want to touch other tests here. This goes beyond the description of this PR. Anyway I will fix it but should be part of another PR.

**Comment by skonto** [comment]
Tests that fail:

```
    tracker.go:139: Waiting for eventing-e2e14 to be deleted
            --- FAIL: TestServerlessUpgrade/PostUpgradeTests/EventingPostUpgradeTest/KafkaChannel-messaging.knative.dev/v1beta1 (302.14s)
```

**Pull request merged**